source/jormungandr/jormungandr/realtime_schedule/cleverage.py

```diff
@@ -0,0 +1,116 @@
+from jormungandr.realtime_schedule.realtime_proxy import RealtimeProxy
+from flask import logging
+import pybreaker
+import pytz
+import requests as requests
+from jormungandr import cache, app
+
+
+class Cleverage(RealtimeProxy):
+    """
+    class managing calls to cleverage external service providing real-time next passages
+    """
+    def __init__(self, id, service_url, service_args, timezone, object_id_tag=None, timeout=10):
+        self.service_url = service_url if (service_url[-1] == u'/') else (service_url+'/')
+        self.service_args = service_args
+        self.timeout = timeout  # timeout in seconds
+        self.rt_system_id = id
+        self.object_id_tag = object_id_tag if object_id_tag else id
+        self.breaker = pybreaker.CircuitBreaker(fail_max=app.config['CIRCUIT_BREAKER_MAX_CLEVERAGE_FAIL'],
+                                                reset_timeout=app.config['CIRCUIT_BREAKER_CLEVERAGE_TIMEOUT_S'])
+        self.timezone = pytz.timezone(timezone)
+
+    def __repr__(self):
+        """
+         used as the cache key. we use the rt_system_id to share the cache between servers in production
+        """
+        return self.rt_system_id
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CLEVERAGE', 30))
+    def _call_cleverage(self, url):
+        """
+        http call to cleverage
+        """
+        try:
+            headers = {}
+            for k, v in self.service_args.items() :
+                headers[k] = v
+            return self.breaker.call(requests.get, url, timeout=self.timeout, headers=headers)
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('Cleverage RT service dead, using base '
+                                              'schedule (error: {}'.format(e))
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('Cleverage RT service timeout, using base '
+                                              'schedule (error: {}'.format(t))
+        except:
+            logging.getLogger(__name__).exception('Cleverage RT error, using base schedule')
+        return None
+
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+
+        if not stop_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug('missing realtime id for {obj}: stop code={s}'.
+                                              format(obj=route_point, s=stop_id))
+            return None
+
+        url = "{base_url}{stop_id}".format(base_url=self.service_url, stop_id=stop_id)
+
+        return url
+
+    def _get_passages(self, route_point, cleverage_resp):
+        logging.getLogger(__name__).debug('cleverage response: {}'.format(cleverage_resp))
+        line = route_point.pb_route.line
+        print line.code
```

Comment:
and may be something should be returned here? (since the function name is `get`)